pyproject.toml

```diff
@@ -56,7 +56,7 @@ pinned = [ # Pinned versions of core dependencies
   'pooch<1.9.0',
   'scooby<0.11.0',
   'typing-extensions<4.13.0',
-  'vtk<9.4.0',
+  'vtk<=9.4.0',
```

Comment:
let's hold off on this until #6731 is fully resolved

Reply:
sounds good! Are you okay if I continue to work on the PR in the meantime, to get it ready to ship?
for eg: I need to do some VTK version checks, to avoid import errors if the user happens to have an old version of VTK

Reply:
Yeah, totally fine to to keep working on this, I may just ask we block merging until VTK 9.4 support is ironed out

Reply:
hey @banesullivan; I've rebased the PR over https://github.com/pyvista/pyvista/pull/6879 to benefit from the fixes you've put in, and to make sure the CI failures I would see are related to the change this PR is trying to introduce.

I've tried to put in some logic around handling different vtk versions but its really not fun trying to run the full CI locally; would it be possible for you to approve a CI run for this branch as it stands, to help me see where any remaining issues are?

Reply:
I approved the workflows, this approval process seems new (they used to run automatically), will try to approve when more pushes are made